File: Java/src/main/java/workshops/testingil/dirtytests/app/CalculatorController.java
package workshops.testingil.dirtytests.app;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.json.JSONObject;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController()
@CrossOrigin()
@RequestMapping("/calc")
public class CalculatorController {

    Calculator calc = new Calculator();


    @PostMapping("/press")
    public void press(@RequestParam String key ){
        System.out.println("-------- received pressed " + key + " -------------------");
        calc.press(key);
    }

    @GetMapping(value = "/display", produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<String> getDisplay() throws JsonProcessingException {
        ResultWrapper wrapper = new ResultWrapper();
        wrapper.setDisplay(calc.getDisplay());
        ObjectMapper mapper = new ObjectMapper();

        String response = mapper.writeValueAsString(wrapper);
        System.out.println("-------- getting display " + response + " -------------------");
        return ResponseEntity.ok().body(response);
    }

    @PostMapping("/sequence")
    public void pressSequence(@RequestBody String sequenceMessage) throws Exception {
        System.out.println("-------- received sequence " + sequenceMessage + " -------------------");
        JSONObject jsonObject = new JSONObject(sequenceMessage);
        calc.pressAll(jsonObject.getString("sequence"));
    }
    @PostMapping("/reset")
    public void reset(){
        System.out.println("-------- received reset -------------------");
        calc.reset();
    }

    @GetMapping("/")
    public String getRoot(){
        return "Working hard";
    }

}

File: C#/NewDirtyTests/DirtyTestsWorkshop/DirtyTestsWorkshop/App/CalculatorController.cs
using DirtyTestsWorkshop.App;
using Newtonsoft.Json.Linq;

namespace DirtyTestsWorkshop.Controllers
{

/*
    [ApiController]
    [Route("/calc")]
    public class CalculatorController : ControllerBase
    {
        readonly Calculator calc;
        public CalculatorController()
        {
            calc = new Calculator();
        }

        [HttpPost("/press")]
        public ActionResult Press(String key)
        {
            Console.WriteLine("-------- received pressed " + key + " -------------------");
            calc.Press(key);
            return Ok();
        }

        [HttpGet("/display")]
        [Produces("application/json")]
        public ActionResult<ResultWrapper> GetDisplay()
        {
            ResultWrapper wrapper = new ResultWrapper();
            wrapper.setDisplay(calc.GetDisplay());
            Console.WriteLine("-------- getting display " + calc.GetDisplay() + " -------------------");
            return new OkObjectResult(wrapper);
        }

        [HttpPost("/sequence")]
        public ActionResult PressSequence([FromBody] string sequenceMessage)
        {
            Console.WriteLine("-------- received sequence " + sequenceMessage + " -------------------");
            JObject json = JObject.Parse(sequenceMessage);
            string sequence = (string)json["sequence"];
            calc.PressAll(sequence);
            return Ok();
        }

        [HttpPost("/reset")]
        public ActionResult Reset()
        {
            Console.WriteLine("-------- received reset -------------------");
            calc.Reset();
            return Ok();
        }


        [HttpGet("/")]
        public ActionResult<string> GetRoot()
        {
            return Ok("Working hard");
        }

    }
*/ 

}
